lib/ecl/ecl_nnc_data.c

```diff
@@ -197,36 +198,37 @@ static void ecl_nnc_data_set_values(ecl_nnc_data_type * data, const ecl_grid_typ
          if (current_kw) {            
             correct_kw_count = ecl_kw_get_size( current_kw );
             check_kw_count = true;
-            data->size = nnc_index + correct_kw_count;
          }
          else {
-            check_kw_count = false;
-            printf("Warning: failed to obtain kw from file in function %s. ", __func__);
-            printf("Grid1: %d, Grid2 %d\n", current_grid1, current_grid2);
+            all_nnc_kw_present = false;
          }
       }
       if (current_kw) {
-         data->values[nnc_index] = ecl_kw_iget_as_double(current_kw, pair->input_index);
-         kw_count++;
+        data->values[nnc_index] = ecl_kw_iget_as_double(current_kw, pair->input_index);
+        kw_count++;
       }
-      else
-         data->values[nnc_index] = -1;
+      
    }
    assert_correct_kw_count(current_kw, __func__, check_kw_count, correct_kw_count, kw_count);
```

Comment:
The `bool check_kw_count` should be removed from `assert_correct_kw_count( )` function - if you call the `assert_xxx` the check should be applied!